src/web/app/pages-instructor/instructor-track-view-page/instructor-track-view-page.component.ts

```diff
@@ -117,10 +118,13 @@ export class InstructorTrackViewPageComponent implements OnInit {
           })),
           mergeAll(),
           finalize(() => this.isLoading = false))
-      .subscribe(((feedbackSession: FeedbackSessions) =>
+      .subscribe(((feedbackSession: FeedbackSessions) => {
+        if (feedbackSession.feedbackSessions.length > 0) {
           this.courseToFeedbackSession[feedbackSession.feedbackSessions[0].courseId]
-            = [...feedbackSession.feedbackSessions]),
-        (e: ErrorMessageOutput) => this.statusMessageService.showErrorToast(e.error.message));
+            = [...feedbackSession.feedbackSessions];
```

Comment:
Just some discussion: do you think there could be the concern where `this.courseToFeedbackSession[<session_A>.courseId]` might be overwritten by another api response with the same course id?

If so, you might want to use the use the `map` `getOrDefault` kind of pattern. But if such concern does not exist, it is fine.

The semantics is entirely up to you.

Reply:
I don't think it will be overwritten since feedback sessions with the same `courseId` will be retrieved together. So all the `feedbackSessions.feedbackSessions[0]` will have different `courseId`

Reply:
Alright!